python/ml4ir/base/model/metrics/metrics_impl.py

```diff
@@ -41,6 +43,9 @@ def get_metrics_impl(
     metrics_impl: List[Union[Metric, str]] = list()
 
     for metric in metrics:
+        if isinstance(metric, ranking_metrics_impl.CombinationMetric) and for_aux_output:
```

Comment:
Here, it's no allowing RankMatchFailure to be a metric for aux_ouput, but it allows the other metrics as (MRR) to be a metric for aux output, right?
It seems to me that MRR and the other non NMF metrics shouldn't be computed for the aux output because they don't convey any info. In fact it's useless to learn the value of train_aux_ranking_score_new_MRR as it's being measured against the title scores which are not clicks (correct me if I'm wrong here) . However, the NMF metric should be part of either the primary output or the aux output (if it's allowed for both, then both should give the same metric value, right? i.e. for example: train_ranking_score_new_RankMatchFailure should be equals to train_aux_ranking_score_new_RankMatchFailure.

Reply:
To keep the implementation straight-forward, combination metrics are defined only for the "main output". The other metrics can/cannot be relevant to the aux-output (depending on the actual aux label). That's the reasoning behind the current design. Let me know if you feel another approach is more appropriate. 

> However, the NMF metric should be part of either the primary output or the aux output (if it's allowed for both, then both should give the same metric value, right? i.e. for example: train_ranking_score_new_RankMatchFailure should be equals to train_aux_ranking_score_new_RankMatchFailure.

This is accurate, but an implementation nightmare at this point. Open to suggestions here

Reply:
In that case, it seems to me that no metrics other than loss should be tracked for the aux_output. and all other metrics loss, MRR, ACR and RankNMF should be tracked for the main output. In other words, the only change from training a model without aux_labels is that we should we should add RankNMF to the set of metrics tracked by the main output whenever aux_label is given.

Reply:
This makes sense for the particular aux feature we're talking about. We might want to measure MRR on a different aux output, no?

Reply:
Generally it's possible, but that aux source has to be clicks, no?
If we want to make this account for all different cases of possible aux targets/outputs, then perhaps we need to make the metrics to be tracked a user input. What do you think?

Reply:
Yes, that's what I was thinking too. Will track this in a follow up